pkg/runner/mounting_runner.go

```diff
@@ -0,0 +1,102 @@
+package runner
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/buildbarn/bb-remote-execution/pkg/proto/configuration/bb_runner"
+	runner_pb "github.com/buildbarn/bb-remote-execution/pkg/proto/runner"
+	"github.com/buildbarn/bb-storage/pkg/filesystem"
+	"github.com/buildbarn/bb-storage/pkg/filesystem/path"
+	"github.com/buildbarn/bb-storage/pkg/util"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+type mountingRunner struct {
+	base        runner_pb.RunnerServer
+	buildDirectory filesystem.Directory
+	mount *bb_runner.MountInfo
+}
+
+// NewMountingRunner is a decorator for Runner
+// that mounts `mount` before running a build action.
+//
+// This decorator can be used for chroot runners
+// that must mount special filesystems into the input root.
```

Comment:
We could also compare it to the clean runner that has generalized pre- and post-command execution.